### PR TITLE
feat(chart): create bare entrypoint to reduce bundle size

### PR DESCRIPTION
**Pull request by Theeraphat-Sorasetsakul**
## Description

Provide new entrypoint of ef-chart for app to optimize size of bundle

Fixes # ([STG-421](https://jira.refinitiv.com/browse/STG-421))

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/elements/src/chart/bare.ts [hunk 1]
```diff
@@ -0,0 +1,620 @@
...
+   * @type {ChartConfiguration}
+   */
+  @property({ type: Object })
+  public config: ChartConfiguration | null = null;
+
+  /**
+   * Html canvas element
+   * @type {HTMLCanvasElement}
+   */
+  @query('canvas')
```
Comment (wsuwt) [inline packages/elements/src/chart/bare.ts]: Can we use `ref` instead ?

Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/bare.ts]: Done.

File: packages/elements/src/chart/elements/chart.ts [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+        }
+      }
+    };
+  }
+
+  /**
+   * Set global configuration of ChartJS
+   * @returns {void}
+   */
+  // TODO: Try and remove the need for global object modification.
```
Comment (wsuwt) [inline packages/elements/src/chart/elements/chart.ts]: Is this TODO still valid ? If it is, is it worth trying out ?

File: packages/elements/src/chart/elements/chart.ts [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+   * @returns {void}
+   */
+  protected decorateColors = (chart: ChartJS): void => {
+    chart.config.data.datasets.forEach((dataset, datasetIndex) => {
+      let colors;
+      let borderColor;
+      let backgroundColor;
+      const isMultipleDatasets = (chart.config.data.datasets.length > 1);
+      // From old requirement, Only line, radar, scatter, polarArea type are opaque backgroundColor
+      switch (dataset.type || this.config?.type) {
```
Comment (wsuwt) [inline packages/elements/src/chart/elements/chart.ts]:
```suggestion
      switch (dataset.type ?? this.config?.type) {
```

Comment (wsuwt) [reply packages/elements/src/chart/elements/chart.ts]: This only check for `null` and `undefined` though.

File: packages/elements/src/chart/elements/chart.ts [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+
+      return legends;
+    }
+
+    if (ChartJS.defaults.plugins.legend.labels.generateLabels) {
+      legends = ChartJS.defaults.plugins.legend.labels.generateLabels(chart);
+    }
+    legends.forEach((legend, i) => {
+      legend.lineWidth = Number(datasets[i].borderWidth) || 0;
+      switch (datasets[i].type || chartType) {
```
Comment (wsuwt) [inline packages/elements/src/chart/elements/chart.ts]:
```suggestion
      switch (datasets[i].type ?? chartType) {
```

File: packages/elements/src/chart/elements/chart.ts (hunk 2 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/elements/chart.ts]: I tried and found that without global config cannot cover our style to creating a new custom feature, for example a new custom scale. And also with global config is easier readability.

Comment (wsuwt) [reply packages/elements/src/chart/elements/chart.ts]: So that means this TODO is not valid anymore. Should we remove it ?

Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/elements/chart.ts]: You're right. I will remove it.

File: packages/elements/src/chart/elements/chart.ts [hunk 5]
```diff
@@ -0,0 +1,603 @@
...
+ */
+declare module 'chart.js' {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  interface PluginOptionsByType<TType extends ChartType> {
+    'ef-chart': object;
+  }
+}
+
+/**
+ * Charting component that use ChartJS library
```
Comment (wsuwt) [inline packages/elements/src/chart/elements/chart.ts]:
```suggestion
 * Charting component that uses ChartJS library
```

File: packages/elements/src/chart/plugins/index.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import doughnutCenterLabelPlugin from './doughnut-center-label.js';
+export {
+  doughnutCenterLabelPlugin
+};
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/plugins/index.ts]:
```suggestion
export * from './doughnut-center-label.js';
```

File: packages/elements/src/chart/plugins/doughnut-center-label.ts
Comment (wattachai-lseg) [inline packages/elements/src/chart/plugins/doughnut-center-label.ts:1]: change `export default plugins;` to `export { plugins as doughnutCenterLabelPlugin };`

File: packages/elements/src/chart/elements/chart.ts [hunk 7]
```diff
@@ -0,0 +1,600 @@
...
+      this.createChart();
+    }
+  }
+
+  /**
+   * Get as CSS variable and tries to convert it into a usable number
+   * @returns The value as a number, or, undefined if NaN.
+   */
+  protected cssVarAsNumber (...args: string[]): number | undefined {
+    const result = Number(this.getComputedVariable(...args).replace(/\D+$/, ''));
```
Comment (wsuwt) [inline packages/elements/src/chart/elements/chart.ts]: Can you double check this security hotspot ? https://sonarcloud.io/project/security_hotspots?id=refinitiv_refinitiv-ui&pullRequest=687&resolved=false&types=SECURITY_HOTSPOT

Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/elements/chart.ts]: We can be continuous on [STG-434](https://jira.refinitiv.com/browse/STG-434).

**Review by wattachai-lseg — APPROVED** [review]

**Pull request merged**